src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexStatementCondition.java

```diff
@@ -0,0 +1,165 @@
+            smallFilter.put(token.toString());
+            mediumFilter.put(token.toString());
+            largeFilter.put(token.toString());
+        });
+
+        final long rowId = StreamDBClient.BloomFiltersTempTable
+                .insert(config.context(), smallFilter, mediumFilter, largeFilter);
+
+        final Condition rowIdCondition = StreamDBClient.BloomFiltersTempTable.id.eq(rowId);
+
+        final Field<byte[]> smallColumn = DSL
+                .select(StreamDBClient.BloomFiltersTempTable.fe100kfp001)
+                .from(StreamDBClient.BloomFiltersTempTable.BLOOM_TABLE)
+                .where(rowIdCondition)
+                .asField();
+        final Field<byte[]> mediumColumn = DSL
+                .select(StreamDBClient.BloomFiltersTempTable.fe1000kfpp003)
+                .from(StreamDBClient.BloomFiltersTempTable.BLOOM_TABLE)
+                .where(rowIdCondition)
+                .asField();
+        final Field<byte[]> largeColumn = DSL
+                .select(StreamDBClient.BloomFiltersTempTable.fe2500kfpp005)
+                .from(StreamDBClient.BloomFiltersTempTable.BLOOM_TABLE)
+                .where(rowIdCondition)
+                .asField();
+
+        final Field<Boolean> fe100kfp001 = DSL
+                .function(
+                        "bloommatch", Boolean.class, smallColumn,
+                        BLOOMDB.FILTER_EXPECTED_100000_FPP_001.FILTER
+                );
+        final Field<Boolean> fe1000kfpp003 = DSL
+                .function(
+                        "bloommatch", Boolean.class, mediumColumn,
+                        BLOOMDB.FILTER_EXPECTED_1000000_FPP_003.FILTER
+                );
+        final Field<Boolean> fe2500kfpp005 = DSL
+                .function(
+                        "bloommatch", Boolean.class, largeColumn,
+                        BLOOMDB.FILTER_EXPECTED_2500000_FPP_005.FILTER
+                );
+
+        final Condition noBloomFilter = BLOOMDB.FILTER_EXPECTED_100000_FPP_001.FILTER
+                .isNull()
+                .and(
+                        BLOOMDB.FILTER_EXPECTED_1000000_FPP_003.FILTER
+                                .isNull()
+                                .and(BLOOMDB.FILTER_EXPECTED_2500000_FPP_005.FILTER.isNull())
+                );
+        final Condition queryCondition = fe100kfp001
+                .eq(true)
+                .or(fe1000kfpp003.eq(true).or(fe2500kfpp005.eq(true).or(noBloomFilter)));
+        LOGGER.trace("ConditionWalker.emitElement bloomCondition part <{}>", queryCondition);
+
+        return queryCondition;
+    }
+
+    @Override
+    public boolean equals(final Object object) {
+        if (this == object) return true;
+        if (object == null) return false;
+        if (object.getClass() != this.getClass()) return false;
+        final IndexStatementCondition cast = (IndexStatementCondition) object;
+        return this.condition().toString().equals(cast.condition().toString());
```

Comment:
compare uses .toString() however toString is not implemented, it must use the object properties that make it equal to another i.e. in this case element and config